File: backend/app_api/domain/entities/Mantenimiento_entity.py
from dataclasses import dataclass
from app_api.domain.values.timestamp import TimestampModel
from datetime import date

@dataclass(kw_only=True)
class MantenimientoEntity(TimestampModel):
    id: int | None
    producto: int
    precio: int
    fecha_programada: date | None
    estado: str
    usuario: int

File: frontend/src/domain/entities/Mantenimiento_entity.ts
import { TimeStamp } from "../values/Timestamp";

export class MantenimientoEntity extends TimeStamp {
    id: number | null;
    producto: number;
    precio: number;
    fecha_programada: string | null;
    estado: string;
    usuario: number;

    constructor(id: number | null, producto: number, precio: number, fecha_programada: string | null, estado: string, usuario: number,
        created_at: Date | null = null,
        updated_at: Date | null = null,
        deleted_at: Date | null = null
    ) {
        super(created_at, updated_at, deleted_at);
        this.id = id;
        this.producto = producto;
        this.precio = precio;
        this.fecha_programada = fecha_programada;
        this.estado = estado;
        this.usuario = usuario;
    }
}

